include/lmms_basics.h

```diff
@@ -125,9 +120,196 @@ constexpr char LADSPA_PATH_SEPERATOR =
 #endif
 
 
+class sampleFrame
```

Comment:
Would it make sense to turn sampleFrame into it's own cpp file?

Reply:
Yes, this is something that can be done once it is clear if this pull request will go in or not. See this comment: https://github.com/LMMS/lmms/pull/7156#issuecomment-2024579623. I am not sure if the branch will still apply but I think it did not take that much effort.

Reply:
> I have a local branch ready where sampleFrame is renamed to SampleFrame and where it's moved into its own file.

Can't you just merge that branch in here? Looks like this PR has a chance of merge.

Reply:
Done via cherry-picks in commits ad65951ccb5 and 017465d61df.